src/lib/rose_tree/rose_tree.ml

```diff
@@ -0,0 +1,19 @@
+open Core_kernel
+
+type 'a t = T of 'a * 'a t list
```

Comment:
Kind of curious why you named your type t like this. Why not use a record?

Reply:
Could have used a record, but I find using a pair neater for this. Tuple types are not allowed to recurse though, so the trick is to a use a scalar ADT.